src/paulscode/android/mupen64plusae/profile/TouchscreenProfileActivity.java

```diff
@@ -385,37 +397,125 @@ private void hideSystemBars()
+                    return false;
+            }
             
-            popupDialog( assetName, title, index );
+            // Get the N64 index of the button that was pressed
+            int index = mTouchscreenMap.getButtonPress( x, y );
+            if( index != TouchMap.UNMAPPED )
+            {
+                dragIndex = index;
+                dragAsset = TouchMap.ASSET_NAMES.get( index );
+                dragFrame = mTouchscreenMap.getButtonFrame( dragAsset );
+            }
+            else
+            {
+                // See if analog was pressed
+                Point point = mTouchscreenMap.getAnalogDisplacement( x, y );
+                int dX = point.x;
+                int dY = point.y;
+                float displacement = FloatMath.sqrt( ( dX * dX ) + ( dY * dY ) );
+                if( mTouchscreenMap.isInCaptureRange( displacement ) )
+                {
+                    dragAsset = ANALOG;
+                    dragFrame = mTouchscreenMap.getAnalogFrame();
+                }
+            }
+            
+            dragX = mProfile.getInt( dragAsset + TAG_X, INITIAL_ASSET_POS );
+            dragY = mProfile.getInt( dragAsset + TAG_Y, INITIAL_ASSET_POS );
+            
+            return true;
         }
-        else
+        else if( ( event.getAction() & MotionEvent.ACTION_MASK ) == MotionEvent.ACTION_MOVE )
         {
-            // See if analog was pressed
-            Point point = mTouchscreenMap.getAnalogDisplacement( x, y );
-            int dX = point.x;
-            int dY = point.y;
-            float displacement = FloatMath.sqrt( ( dX * dX ) + ( dY * dY ) );
-            if( mTouchscreenMap.isInCaptureRange( displacement ) )
-                popupDialog( ANALOG, getString( R.string.controller_analog ), -1 );
+            if ( dragIndex != TouchMap.UNMAPPED || ANALOG.equals(dragAsset) )
+            {
+                if ( !dragging )
+                {
+                    int dX = x - initialX;
+                    int dY = y - initialY;
+                    float displacement = FloatMath.sqrt( ( dX * dX ) + ( dY * dY ) );
+                    if ( displacement >= 10 )
+                        dragging = true;
+                }
+                if ( !dragging )
+                    return false;
+                
+                // drag this button or analog stick around
+                
+                // calculate the X and Y percentage
+                View view = getWindow().getDecorView();
+                int newDragX = ( x - ( initialX - dragFrame.left ) ) * 100/( view.getWidth() - ( dragFrame.right - dragFrame.left ) );
+                int newDragY = ( y - ( initialY - dragFrame.top ) ) * 100/( view.getHeight() - ( dragFrame.bottom - dragFrame.top ) );
+                
+                // round to the nearest 5%
+                newDragX = Math.min( Math.max( Math.round( newDragX / 5 ) * 5, 0 ), 100 );
```

Comment:
You may have noticed that the current UI with the sliders does allow you to increment by 1% by tapping the +/- keys on either side of the slider.  Would be nice if we could let the user nudge with the new drag system.  Not sure how though without adding a lot of complexity.  Open to ideas...